src/vcpkg/commands.format-feature-baseline.cpp

```diff
@@ -0,0 +1,92 @@
+#include <vcpkg/base/checks.h>
+#include <vcpkg/base/files.h>
+
+#include <vcpkg/commands.format-feature-baselinet.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+
+using namespace vcpkg;
+
+namespace
+{
+    bool is_comment(const std::string& line)
+    {
+        auto index = line.find_first_not_of(" \t");
+        if (index != std::string::npos)
+        {
+            return line[index] == '#';
+        }
+        return true;
+    }
+
+    // The file should only contain ascii characters. Hardcode the unicode collation order for the ascii range.
+    constexpr uint8_t comparison_indices[] = {
+        0,   1,   2,   3,   4,   5,   6,   7,   8,   9,   10,  11,  12,  13,  14,  15,  16,  17,  18,  19,  20,  21,
+        22,  23,  24,  25,  26,  27,  28,  29,  30,  31,  32,  38,  42,  54,  64,  55,  53,  41,  43,  44,  50,  58,
+        35,  34,  40,  51,  65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  37,  36,  59,  60,  61,  39,  49,  75,
+        77,  79,  82,  84,  86,  88,  90,  92,  94,  96,  98,  100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120,
+        122, 124, 126, 45,  52,  46,  57,  33,  56,  76,  78,  80,  81,  83,  85,  87,  89,  91,  93,  95,  97,  99,
+        101, 103, 105, 107, 109, 111, 113, 115, 117, 119, 121, 123, 125, 47,  62,  48,  63};
+
+    struct cmp_char
+    {
+        auto operator()(char l_, char r_)
+        {
+            auto l = static_cast<uint8_t>(l_);
+            auto r = static_cast<uint8_t>(r_);
+            if (l >= sizeof(comparison_indices) || r >= sizeof(comparison_indices))
+            {
+                return l < r;
+            }
+            return comparison_indices[l] < comparison_indices[r];
+        }
+    };
+
+    struct cmp_str
+    {
+        auto operator()(const std::string& left, const std::string& right)
+        {
+            return std::lexicographical_compare(left.begin(), left.end(), right.begin(), right.end(), cmp_char{});
+        }
+    };
+}
+
+namespace vcpkg
+{
+    constexpr CommandMetadata CommandFormatFeatureBaselineMetadata = {
+        "format-feature-baseline",
+        msgCmdFormatFeatureBaselineSynopsis,
+        {msgCmdFormatFeatureBaselineExample},
+        Undocumented,
+        AutocompletePriority::Public,
+        1,
+        1,
+        {{}, {}, {}},
+        nullptr,
+    };
+
+    void command_format_feature_baseline_and_exit(const VcpkgCmdArguments& args, const Filesystem& fs)
```

Comment:
2 problems with this:

1. The assumption that a block starts with a commented line isn't necessarily true. Sometimes, the comment needs to follow the block. I think the existence of this thing is probably worth including even if we don't have a great solution to it though. Clearly this one wasn't in the right place before either:
  ![missorting example of ms-gltf](https://github.com/user-attachments/assets/e34326b5-8396-466f-be1c-883113142e35)
2. This doesn't actually lexically sort port names. I think we can fudge this by just sorting : earlier? 
  ![misssorting example cmake and cmake-user](https://github.com/user-attachments/assets/30bcff06-a191-480f-adb7-03a8c2218f31)

Reply:
I fixed this and applied to vcpkg's current registry here: https://github.com/microsoft/vcpkg/pull/44136/

Reply:
This command was never meant to format `ci.baseline.txt`. It created it to format https://github.com/autoantwort/vcpkg/blob/test-features/scripts/ci.feature.baseline.txt

Reply:
> This doesn't actually lexically sort port names. I think we can fudge this by just sorting : earlier?

Yeah the new sorting order is better 👍